src/wp-admin/async-upload.php

```diff
@@ -133,6 +133,7 @@
 			'paragraph_wrap'     => false,
 		)
 	);
+	echo "<script type=\"text/javascript\">jQuery( 'button.dismiss' ).on( 'click', function() {jQuery(this).parents('div.media-item').slideUp(200, function(){jQuery(this).remove();})});</script>\n";
```

Comment:
When uploading multiple images , this results in n click handlers on the first failed upload, n-1 on the second and so on. 

![Screenshot 2024-10-07 at 2 17 57 PM](https://github.com/user-attachments/assets/b720bdd6-b5ab-48fd-a8b7-1200125460fc)

I think it will work if you're more specific about the element the click handler is added to: `jquery( this ).parents('div.media-item').find('button.dismiss')` [untested, of course].

Is there any risk of a memory leak as the elements are removed or does jQuery clean up after itself?

Reply:
According to the docs on `.remove()`, jQuery removes all bound events and associated data when remove is called, so I think we should be safe from memory leaks here.